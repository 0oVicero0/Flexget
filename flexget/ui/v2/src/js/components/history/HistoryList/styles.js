import styled from 'emotion/react';
import { ListSubheader } from 'material-ui/List';

export const Subheader = styled(ListSubheader)`
  background-color: white;
`;

export const Wrapper = styled.div`
  overflow-y: scroll;
  height: calc(100% + 2rem);
  margin: -1rem;
`;
