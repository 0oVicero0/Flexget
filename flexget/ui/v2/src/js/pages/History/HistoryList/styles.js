import styled from 'emotion/react';
import { ListSubheader } from 'material-ui/List';

export const Subheader = styled(ListSubheader)`
  background-color: white;
`;

export const Wrapper = styled.div`
  overflow-y: scroll;
  flex: 1;
`;
