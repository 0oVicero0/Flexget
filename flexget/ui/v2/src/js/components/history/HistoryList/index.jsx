import React, { Component } from 'react';
import PropTypes from 'prop-types';
import InfiniteScroll from 'react-infinite-scroller';
import List, { ListItem, ListItemText } from 'material-ui/List';
import LoadingSpinner from 'components/common/LoadingSpinner';
import { Subheader, Wrapper } from './styles';

class HistoryList extends Component {
  static propTypes = {
    grouping: PropTypes.oneOf(['time', 'task']).isRequired,
    history: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.shape({
      task: PropTypes.string,
      title: PropTypes.string,
      url: PropTypes.string,
      filename: PropTypes.string,
      details: PropTypes.string,
      time: PropTypes.strings,
      id: PropTypes.number,
    }))).isRequired,
    hasMore: PropTypes.bool,
    getHistory: PropTypes.func.isRequired,
  };

  static defaultProps = {
    hasMore: true,
  };

  render() {
    const { grouping, history, hasMore, getHistory } = this.props;

    return (
      <Wrapper>
        <InfiniteScroll
          hasMore={hasMore}
          loadMore={getHistory}
          loader={<LoadingSpinner loading />}
          ref={(node) => { this.scroller = node; }}
          useWindow={false}
        >
          { Object.entries(history).map(([subheader, histories]) => (
            <List key={subheader} subheader={<Subheader color="primary">{subheader}</Subheader>}>
              {histories.map(({ id, title, time, task }) => (
                <ListItem key={id}>
                  <ListItemText primary={title} secondary={grouping === 'time' ? task : time} />
                </ListItem>
              ))}
            </List>
          ))}
        </InfiniteScroll>
      </Wrapper>
    );
  }
}

export default HistoryList;
