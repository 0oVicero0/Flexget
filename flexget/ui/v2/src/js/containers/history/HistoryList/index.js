import { connect } from 'react-redux';
import HistoryList from 'components/history/HistoryList';
import { getGroupedHistory } from 'store/history/selectors';

export function mapStateToProps({ history }, { grouping }) {
  return {
    history: getGroupedHistory(history, grouping),
    hasMore: !history.totalCount || history.items.length < history.totalCount,
  };
}

export default connect(mapStateToProps)(HistoryList);
