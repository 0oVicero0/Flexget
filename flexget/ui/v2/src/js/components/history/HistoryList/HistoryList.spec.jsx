import React from 'react';
import { shallow } from 'enzyme';
import HistoryList from 'components/history/HistoryList';

describe('components/history/HistoryList', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <HistoryList
        getHistory={jest.fn()}
        history={{
          '2017-09-09': [{
            task: 'task',
            id: 1,
            title: 'something',
          }],
        }}
        grouping="time"
        hasMore
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
