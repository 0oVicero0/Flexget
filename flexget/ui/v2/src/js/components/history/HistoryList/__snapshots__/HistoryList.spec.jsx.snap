// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/history/HistoryList renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled(Component)>
    <InfiniteScroll
        element="div"
        hasMore={true}
        initialLoad={true}
        isReverse={false}
        loadMore={[Function]}
        loader={
            <LoadingBar
                loading={true}
            />
        }
        pageStart={0}
        threshold={250}
        useCapture={false}
        useWindow={false}
    >
        <withStyles(List)
            subheader={
                <styled(withStyles(ListSubheader))
                    color="primary"
                >
                    2017-09-09
                </styled(withStyles(ListSubheader))>
            }
        >
            <withStyles(ListItem)>
                <withStyles(ListItemText)
                    primary="something"
                    secondary="task"
                />
            </withStyles(ListItem)>
        </withStyles(List)>
    </InfiniteScroll>
</styled(Component)>,
  "nodes": Array [
    <styled(Component)>
      <InfiniteScroll
            element="div"
            hasMore={true}
            initialLoad={true}
            isReverse={false}
            loadMore={[Function]}
            loader={
                  <LoadingBar
                        loading={true}
                  />
            }
            pageStart={0}
            threshold={250}
            useCapture={false}
            useWindow={false}
      >
            <withStyles(List)
                  subheader={
                        <styled(withStyles(ListSubheader))
                              color="primary"
                        >
                              2017-09-09
                        </styled(withStyles(ListSubheader))>
                  }
            >
                  <withStyles(ListItem)>
                        <withStyles(ListItemText)
                              primary="something"
                              secondary="task"
                        />
                  </withStyles(ListItem)>
            </withStyles(List)>
      </InfiniteScroll>
</styled(Component)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <HistoryList
        getHistory={[Function]}
        grouping="time"
        hasMore={true}
        history={
                Object {
                        "2017-09-09": Array [
                          Object {
                            "id": 1,
                            "task": "task",
                            "title": "something",
                          },
                        ],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": HistoryList {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "getHistory": [Function],
          "grouping": "time",
          "hasMore": true,
          "history": Object {
            "2017-09-09": Array [
              Object {
                "id": 1,
                "task": "task",
                "title": "something",
              },
            ],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled(Component)>
          <InfiniteScroll
                    element="div"
                    hasMore={true}
                    initialLoad={true}
                    isReverse={false}
                    loadMore={[Function]}
                    loader={
                              <LoadingBar
                                        loading={true}
                              />
                    }
                    pageStart={0}
                    threshold={250}
                    useCapture={false}
                    useWindow={false}
          >
                    <withStyles(List)
                              subheader={
                                        <styled(withStyles(ListSubheader))
                                                  color="primary"
                                        >
                                                  2017-09-09
                                        </styled(withStyles(ListSubheader))>
                              }
                    >
                              <withStyles(ListItem)>
                                        <withStyles(ListItemText)
                                                  primary="something"
                                                  secondary="task"
                                        />
                              </withStyles(ListItem)>
                    </withStyles(List)>
          </InfiniteScroll>
</styled(Component)>,
        "_debugID": 2,
        "_renderedOutput": <styled(Component)>
          <InfiniteScroll
                    element="div"
                    hasMore={true}
                    initialLoad={true}
                    isReverse={false}
                    loadMore={[Function]}
                    loader={
                              <LoadingBar
                                        loading={true}
                              />
                    }
                    pageStart={0}
                    threshold={250}
                    useCapture={false}
                    useWindow={false}
          >
                    <withStyles(List)
                              subheader={
                                        <styled(withStyles(ListSubheader))
                                                  color="primary"
                                        >
                                                  2017-09-09
                                        </styled(withStyles(ListSubheader))>
                              }
                    >
                              <withStyles(ListItem)>
                                        <withStyles(ListItemText)
                                                  primary="something"
                                                  secondary="task"
                                        />
                              </withStyles(ListItem)>
                    </withStyles(List)>
          </InfiniteScroll>
</styled(Component)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <HistoryList
    getHistory={[Function]}
    grouping="time"
    hasMore={true}
    history={
        Object {
            "2017-09-09": Array [
              Object {
                "id": 1,
                "task": "task",
                "title": "something",
              },
            ],
          }
    }
/>,
}
`;
